source/pinia/statistics-store.ts

```diff
@@ -14,17 +14,80 @@
 
 import { type Stats } from '@providers/stats'
 import { defineStore } from 'pinia'
-import { ref } from 'vue'
+import { anyLast30Days, avgAny30Days, sumAny30Days } from 'source/common/util/stats'
+import { DateTime } from 'luxon'
+import { ref, computed } from 'vue'
 
 const ipcRenderer = window.ipc
 
+/**
+ * Generates the past 30 days as ISO dates in the form YYYY-MM-DD
+ *
+ * @return  {string[]}  The date strings
+ */
+function generateLast30ISODates (): string[] {
+  const dates: string[] = []
+
+  let currentDate = DateTime.now()
+
+  while (dates.length < 30) {
+    dates.push(currentDate.toISODate())
+    currentDate = currentDate.minus({ days: 1 })
+  }
+
+  return dates
+}
+
+function anyLast30CalendarDays (record: Array<[string, number]>): Array<[string, number]> {
```

Comment:
The names `anyLast30CalendarDays` and `anyLast30Days` are very similar at first sight, yet the functions are doing different things.

We could use the following names instead:
- `anyLast30Days` -> `last30EntriesByDate`
- `anyLast30CalendarDays` -> `alignToLast30CalendarDays`

PS: I'm not that creative with function names, so guess who I asked for advice ;)

Reply:
> PS: I'm not that creative with function names

Who is, really? ;) But I think your suggestions make perfect sense, and say a bit more about what the functions really do, so thanks for the suggestions! I've implemented them.